### PR TITLE
Feat(CX-3358): MoneyInput Improvements

### PR DESCRIPTION
**Pull request by lordkiz**
This PR resolves [CX-3358] 
Co-Authored: @olerichter00 

### Description

This PR fixes:
- The inability to have the required fontsize in MoneyInput and PhoneInput
- Allows for formatting money in a en-US readable format in the input
- Money format allows cents to a maximum of 2 decimal places

https://user-images.githubusercontent.com/18648835/216303785-05c59fdf-5457-4c23-902a-b0000b13381e.mov

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- MoneyInput Improvements - kizito

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3358]: https://artsyproduct.atlassian.net/browse/CX-3358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/store/config/features.ts [hunk 1]
```diff
@@ -204,6 +204,11 @@ export const features = defineFeatures({
     showInDevMenu: true,
     readyForRelease: false,
   },
+  AREnableMoneyFormattingInMyCollectionForm: {
+    description: "Enable Money formatting in MyCollection Form",
+    showInDevMenu: true,
+    readyForRelease: false,
```
Comment (olerichter00) [inline src/app/store/config/features.ts:210]: **question:** Shall we add an echo flag to be able to release the feature without a new app release?

File: src/palette/elements/Input/MoneyInput/MoneyInput.tsx [hunk 2]
```diff
@@ -150,3 +160,43 @@ const currencyOptions: Array<SelectOption<SupportedCurrencies>> = [
   { label: "EUR €", value: "EUR" },
   { label: "GBP £", value: "GBP" },
 ]
+
+// Money Formatting
```
Comment (olerichter00) [inline src/palette/elements/Input/MoneyInput/MoneyInput.tsx:164]: **suggestion:** Maybe we can move these helpers into another file (e.g. `moneyInputHelpers.ts`).

**Review by olerichter00 — COMMENTED** [review]
I've tested the component in the simulator and there is a quite long lag between typing and formating the input (about 1 second). Is this expected? Or is it maybe just because the simulator is too slow?

The font size of the currency symbol seems to be smaller than the amount. Is this expected?
Screenshot 2023-02-02 at 12 35 18

File: src/app/store/config/features.ts (hunk 1 above)
Comment (lordkiz) [reply src/app/store/config/features.ts:210]: Sure

File: src/palette/elements/Input/MoneyInput/MoneyInput.tsx (hunk 2 above)
Comment (lordkiz) [reply src/palette/elements/Input/MoneyInput/MoneyInput.tsx:164]: > I've tested the component in the simulator and there is a quite long lag between typing and formating the input (about 1 second). Is this expected? Or is it maybe just because the simulator is too slow?
> 
> The font size of the currency symbol seems to be smaller than the amount. Is this expected? Screenshot 2023-02-02 at 12 35 18

I think a small lag is inevitable if one types really fast on a laptop keyboard when formatting strings. It seems fine on mine when I enable the simulator keyboard `CMD+K`. But will look again.
The ticket specified that currency be the same size as the Input titles i.e fontsize: 13 or xs. Previously it was the same size as input text.

**Review by MrSltun — COMMENTED** [review]
Looks good to me! 🌟 Great work Kizito!! 🙌
I only have the same concern from Ole about the user experience, which is about the time between showing the user input and formatting it (adding the `,`) is there a way we could do it without the time gap?

File: src/palette/elements/Input/MoneyInput/MoneyInput.tsx (hunk 2 above)
Comment (olerichter00) [reply src/palette/elements/Input/MoneyInput/MoneyInput.tsx:164]: Alright, let's give it a try on a real device.

**Comment by lordkiz** [comment]
@olerichter00 @MrSltun This looks pretty fast on my end. Even on a computer keyboard.
There probably can be areas of optimisation

https://user-images.githubusercontent.com/18648835/216359787-5d88188b-f220-4b04-8fe9-0fbefd73a3d4.mov

**Comment by lordkiz** [comment]
@olerichter00 @MrSltun I will check for areas where I can optimise it, see if it can be any faster.

**Comment by lordkiz** [comment]
@olerichter00 @MrSltun Can't find any areas of optimising this. All approaches I've explored are not any faster.

**Comment by olerichter00** [comment]
You could move the value state to the inner component and update the form state in a `useEffect`. I've just tested it and the UI updates are instant ([Updated INTERNALSelectAndInputCombinationBase.tsx](https://gist.github.com/olerichter00/ba6c84e63dc00fd50fe7306b6b4c8441)).

**Comment by lordkiz** [comment]
Thanks @olerichter00, that was the trick.

**Review by olerichter00 — APPROVED** [review]

File: src/app/Components/Bidding/Screens/PhoneNumberForm.tests.tsx [hunk 3]
```diff
@@ -26,9 +26,10 @@ describe("PhoneNumberForm component", () => {
     const phoneInput = getByTestId("phone-input")
     fireEvent.changeText(phoneInput, phoneNumber)
 
-    fireEvent.press(container.queryAllByText("Add phone number")[1])
-
-    expect(onSubmitMock).toHaveBeenLastCalledWith(formattedPhoneNumber)
+    act(() => fireEvent.press(container.queryAllByText("Add phone number")[1]))
```
Comment (pvinis) [inline src/app/Components/Bidding/Screens/PhoneNumberForm.tests.tsx:29]: very late here but act is not needed here. fireevent has act inside it

File: src/app/Components/Bidding/Screens/PhoneNumberForm.tests.tsx [hunk 4]
```diff
@@ -26,9 +26,10 @@ describe("PhoneNumberForm component", () => {
...
+    waitFor(() => {
```
Comment (pvinis) [inline src/app/Components/Bidding/Screens/PhoneNumberForm.tests.tsx:30]: the { here make this invalid. waitfor needs the expect returned directly

**Review by pvinis — COMMENTED** [review]
checking here after inc-167

File: src/app/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tsx [hunk 5]
```diff
@@ -320,7 +320,7 @@ export const updateArtwork = async (
 
   let pricePaidCents
   if (pricePaidDollars && !isNaN(Number(pricePaidDollars))) {
-    pricePaidCents = Number(pricePaidDollars) * 100
```
Comment (pvinis) [inline src/app/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tsx:323]: why this change?
